.scripts/lib.sh

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2021 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+#
+
+
+# -----------------------------------------------------------------------------
+#  FUNCTIONS & PROCEDURES
+# -----------------------------------------------------------------------------
+
+function die() {
+    echo "$@" 1>&2 ; exit 1;
+}
+
+# Given a tag and a module name,
+#   if the tag is a release tag for module, echoes a tag
+#   else returns 1
+# Uses shopt and sets locale, so runs in subshell
+function is_release_tag() (
+    [ "$#" -eq 2 ] || die "2 arguments required, $# provided to 'func:is_release_tag'"
+    local __tag="$1"
+    local __module="$2"
+
+    # For extended glob patterns and ensure [0-9] range
+    shopt -s extglob
+    LC_ALL=C
+
+    case "${__module}" in
+        testing/micro)
+            tag_pattern='micro-+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag#micro-}" || return 1
+            ;;
+        testing/mini)
+            tag_pattern='+([0-9]).+([0-9]).+([0-9])/@(aws|gcp)/@(large|xlarge|xxlarge)'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag%%/*}" || return 1
+            ;;
+        1-trackers/golang-tracker)
+            tag_pattern='v+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag#v}" || return 1
+            ;;
+        4-storage/snowplow-rdb-loader)
+            tag_pattern='r+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag}" || return 1
+            ;;
+        5-data-modeling/data-models)
+            tag_pattern='**/+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag}" || return 1
+            ;;
```

Comment:
Great that you checked each individual repo. We need to unify our tagging style across all the repos.